### PR TITLE
Support service generation on action folders

### PR DESCRIPTION
**Pull request by hidmic**
Connected to ros2/rosidl#301. This pull request is part of a series of pull requests that make the necessary changes to allow us to build .srv files outside an `srv` folder.

connects to ros2/rosidl#301

### CODE REVIEW

File: rosidl_typesupport_c/resource/srv__type_support.cpp.em [hunk 1]
```diff
@@ -94,7 +97,7 @@ static const rosidl_service_type_support_t @(spec.srv_name)_service_type_support
 
 }  // namespace rosidl_typesupport_c
 
-}  // namespace srv
+}  // namespace @(subfolder)
 
 }  // namespace @(spec.pkg_name)
```
Comment (jacobperron) [inline rosidl_typesupport_c/resource/srv__type_support.cpp.em]: I think L105 should use `@(subfolder)`.

https://github.com/ros2/rosidl_typesupport/blob/24d917c5fea5a1a46400c9d026c032bf42a49cd0/rosidl_typesupport_c/resource/srv__type_support.cpp.em#L105

should be

```
#include "@(spec.pkg_name)/@(subfolder)/@(get_header_filename_from_msg_name(spec.srv_name))__@(list(type_supports)[0]).h"
```

**Comment by jacobperron** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/5338/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2073/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/4436/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/5306/)

**Comment by hidmic** [comment]
@jacobperron I believe CI is failing because it's not pulling the associated `rosidl_typesupport_fastrtps` PR'd branch. Does it do any form of branch name matching to fetch sources? If so, we can make it uniform across repos.

**Comment by apojomovsky** [comment]
Ok, given that we have green light I'm proceeding to merge this and its related PRs

**Pull request merged**